---
title: "Assignment 1"
output: html_notebook
---

```{r}
library(tidyverse)
library(readr)
```

### Question) SCN2A mutation- VARIANT_CLASS중 SNV의 Consequence 빈도

# 1. Load Data
```{r}
d = read.delim('table.scn2a.vep_20190916.filtered.txt', sep = '' , header = T)
```

# 2. Plot

이 data의 SCN2A patients는 variant class를 기준으로 deletion, indel, insertion, SNV(single nucleotide variants), substitution의 다섯으로 분류할 수 있고, Consequence type을 기준으로 frameshift variant, inframe deletion, inframe insertion, missense variant, synonymous variant, stop gained, missense variant&splice region variant, stop lost 이렇게 여덟 분류로 나눌 수 있다. 주어진 SCN2A data를 통해 variant class와 consequence type 사이에 연관성이 있는지, 있다면 어떠한 양상을 보이는지 보고자 한다.
```{r}
#Check the column "Consequence", "VARIANT_CLASS" by using the function 'table()'
table(d$Consequence, d$VARIANT_CLASS)
```

실제로 `table()`을 이용하면 SNV와 missense variant가 가장 높은 것을 확인할 수 있다. 특히 SNV의 경우가 전체의 약 98%를 차지하기에 나머지 데이터를 SNV와 같은 항목으로 정리하여 plot을 만드는 것은 비교하기에 부적절해 보인다. 그렇기에 variant class 중 SNV로 한정하여 plot을 만들기로 한다.
```{r}
#Make a data about "SNV" by using tidyverse
D1 <- d %>% filter(grepl("SNV",VARIANT_CLASS)==TRUE) %>% select(Consequence) %>% group_by(Consequence) %>% count()

#Because we don't have "frameshift_variant","inframe_deletion","inframe_insertion" in "SNV" type, we might get the graph with only 5 bar. To compare more corretly, make a new data.frame.
d_plus <- c("frameshift_variant","inframe_deletion","inframe_insertion")
d_plus__ <- data.frame(Consequence = as.factor(d_plus), n = 0)

#Add new data.frame to the first data and draw a plot
bind_rows(D1, d_plus__) %>% complete(fill=list(n=0)) %>%
  ggplot(aes(x=reorder(Consequence, n), y = n, fill=Consequence))+
  geom_bar(stat="identity", width=.6)+
  geom_text(aes(label=n), vjust=1.6, size=3.5)+
  scale_x_discrete(labels=c("missense_variant"="missense","synonymous_variant"="synonymous","stop_gained"="stop gained","missense_variant,splice_region_variant"="missense &\nsplice region","stop_lost"="stop lost","inframe_insertion"="inframe\ninsertion","inframe_deletion"="inframe\ndeletion","frameshift_variant"="frameshift"))+
  guides(fill=FALSE)+
  theme_minimal()+
  coord_flip()+
  xlab("")+
  ylab("SNV")+
  theme_bw()+
  ggtitle("Consequence type of SCN2A patients: especially, SNV")
```

# 3. Explanation

Define the words we saw above first: 

deletion: The point at which one or more contiguous nucleotides were excised.

indel: an insertion or deletion of bases in the genome of an organism.

insertion: The sequence of one or more nucleotides added between two adjacent nucleotides in the sequence.

SNV: SNVs are single nucleotide positions in genomic DNA at which different sequence alternatives exist.

substitution: A sequence alteration where the length of the change in the variant is the same as that of the reference.

missense variant('missense' in plot) : A genetic alteration in which a single base pair substitution alters the genetic code in a way that produces an amino acid that is different from the usual amino acid at that position. Some missense variants (or mutations) will alter the function of the protein. Also called missense mutation.

synonymous variant('synonymous' in plot) : A sequence variant where there is no resulting change to the encoded amino acid.

frameshift variant('frameshift' in plot) :A sequence variant which causes a disruption of the translational reading frame, because the number of nucleotides inserted or deleted is not a multiple of three.

inframe deletion : An inframe non synonymous variant that deletes bases from the coding sequence.

inframe insertion : An inframe non synonymous variant that inserts bases into in the coding sequence.

stop lost : A sequence variant where at least one base of the terminator codon (stop) is changed, resulting in an elongated transcript.

stop gained : A sequence variant whereby at least one base of a codon is changed, resulting in a premature stop codon, leading to a shortened polypeptide.

splice region variant : A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron.

As we can see in the plot, 'missense variant' accounts for the largest proportion in total, also it is about twice more than 'synonymous', which accounts for the second proportion.(Actually, 'missense variant' plus 'missense variant, splicing region variant' is the total 'missense variant' frequency, the value is 780.) That's because this plot was made by using the 'SNV' data. When we translate the mRNA, the third base is less important than the other two, so if there is a change at the third base of the mRNA, it might result in the same amino acid. And there is no reported data on 'inframe insertion', 'inframe deletion', 'frameshift variant'. Because 'inframe insertion' can be founded in the type 'insertion', 'inframe deletion' can be founded in 'deletion', and 'frameshift variant' can be founded in other type of variation, except 'SNV'.